src/lib/parser.c

```diff
@@ -603,14 +603,31 @@ static RdbStatus parserMainLoop(RdbParser *p) {
 
     if (unlikely(p->debugData)) {
         while (1) {
-            RDB_log(p, RDB_LOG_DBG, "[Opcode=%d] %s(State=%d)",
-                    p->currOpcode,
-                    peInfo[p->parsingElement].funcname,
-                    p->elmCtx.state);
+            char buff[500];
+
+            int isNewOpcode = (p->parsingElement == PE_NEXT_RDB_TYPE) ? 1 : 0;
+            int isNewDb = (p->parsingElement == PE_SELECT_DB) ? 1 : 0;
+
+            size_t bytesReadBefore = p->bytesRead;
+            const char *funcName = peInfo[p->parsingElement].funcname;
+            int stateBefore = p->elmCtx.state;
+
             status = peInfo[p->parsingElement].func(p);
-            RDB_log(p, RDB_LOG_DBG, "Return status=%s next %s(State=%d)\n", getStatusString(status),
-                    peInfo[p->parsingElement].funcname,
-                    p->elmCtx.state);
+
+            int at = snprintf(buff, sizeof(buff)-1, "[0x%08zx] %s(State=%d)", bytesReadBefore, funcName, stateBefore);
+
+            /* print important milestones in parsing */
+            if (isNewOpcode)
+                at += snprintf(buff+at, sizeof(buff)-1-at, " [Opcode=%03d]", p->currOpcode);
+            if (isNewDb)
+                at += snprintf(buff+at, sizeof(buff)-1-at, " [SelectDB=%d]", p->selectedDb);
+            if(p->currKeyDbg[0] != '\0')
+                at += snprintf(buff+at, sizeof(buff)-1-at, " [key=%s]", p->currKeyDbg);
+            if (status != RDB_STATUS_OK)
+                at += snprintf(buff+at, sizeof(buff)-1-at, " >>> [Status=%s]", getStatusString(status));
```

Comment:
i suppose all these printfs should be put inside some block that checks the loglevel. i.e. avoid the extra work when not needed.

Reply:
that probably applies to the one in handleNewKeyPrintDbg too.
btw, maybe it should add ellipsis instead of just truncating the key name, that can be confusing.

Reply:
ok, i missed that.